Update map inits to account for Facing -> WAngle

pchote opened the pull request:
With all the plumbing changes out of the way, this is the first of two PRs to wrap up the facing rework by converting mod/map yaml to WAngle.

I am filing this PR now for visibility, but it currently depends on #18271 and #18264. Only the last 3 commits are new here.
I will hold off running the update rules until closer to merging to reduce conflicts if we merged any of @Smittytron's missions before then.

pchote committed:
Remove TurretFacingsInit and PlugsInit.

pchote commented:
Rebased. I'd appreciate if we could get at least one :+1: on the code before I go through the pain of running the updater and then working out which oramap files need to be kept or unstaged.

abcdefg30 commented on OpenRA.Mods.Common/Scripting/Global/ActorGlobal.cs:
```diff
@@ -75,18 +75,17 @@ ActorInit CreateInit(string initName, LuaValue value)
 			}
 
-			// HACK: Forward compatibility for future WAngle facings
+			// HACK: Backward compatibility for legacy WAngle facings
```
`legacy int facings`?

abcdefg30 commented on OpenRA.Mods.Common/Traits/Turreted.cs:
```diff
@@ -30,37 +30,27 @@ public class TurretedInfo : PausableConditionalTraitInfo, Requires<BodyOrientati
 		IEnumerable<ActorInit> IActorPreviewInitInfo.ActorPreviewInits(ActorInfo ai, ActorPreviewType type)
 		{
-			yield return new TurretFacingInit(this, PreviewFacing);
+			// TODO: Update rule needs to account for PreviewFacing removal!
```
Can't this be removed now?

abcdefg30 commented on OpenRA.Mods.Common/Traits/Turreted.cs:
```diff
@@ -30,37 +30,27 @@ public class TurretedInfo : PausableConditionalTraitInfo, Requires<BodyOrientati
 		IEnumerable<EditorActorOption> IEditorActorOptions.ActorOptions(ActorInfo ai, World world)
 		{
-			yield return new EditorActorSlider("Turret", EditorTurretFacingDisplayOrder, 0, 255, 8,
+			yield return new EditorActorSlider("Turret", EditorTurretFacingDisplayOrder, 0, 1024, 8,
```
The old value inclines me to ask if this doesn't want to be `1023`.

abcdefg30 commented on OpenRA.Mods.Common/Traits/ThrowsParticle.cs:
```diff
@@ -66,20 +67,28 @@ public ThrowsParticle(ActorInitializer init, ThrowsParticleInfo info)
+			direction = 1;
+			if (rotationSpeed < 0)
+			{
+				rotation = -rotation;
```
I don't get this here: Why do you invert `rotation` here if it is getting reassign below regardless? (Mabye wants to be `rotationSpeed`?)

pchote committed:
Convert (Dynamic)FacingInit, (Dynamic)TurretFacingInit to WAngle.

pchote commented:
Updated. Fixed @abcdefg30's nits and updated TD and D2k maps. TS map updates will be handled in #18310 to avoid a rebase nightmare.

Holding off updating the RA maps until https://github.com/OpenRA/OpenRA/pulls/Smittytron have either been merged or explicitly marked as depending on this PR.

obrakmann commented:
Smitty's RA maps have now all been merged, removing the dependencies label.

pchote committed:
Update map inits in RA/TD/D2k.

pchote commented:
Updated. This should now be ready to go.

abcdefg30 commented:
But needs a rebase.

abcdefg30 commented on OpenRA.Mods.Common/Traits/Mobile.cs:
```diff
@@ -130,38 +130,13 @@ public bool CanStayInCell(World world, CPos cell)
 		IEnumerable<EditorActorOption> IEditorActorOptions.ActorOptions(ActorInfo ai, World world)
 		{
-			yield return new EditorActorSlider("Facing", EditorFacingDisplayOrder, 0, 255, 8,
+			yield return new EditorActorSlider("Facing", EditorFacingDisplayOrder, 0, 1024, 8,
```
Different place, same problem: `1023`

abcdefg30 commented on OpenRA.Mods.Common/Traits/ThrowsParticle.cs:
```diff
@@ -66,20 +67,29 @@ public ThrowsParticle(ActorInitializer init, ThrowsParticleInfo info)
+			{
+				// WAngle requires positive inputs, so track the direction separately
+				rotationSpeed = -rotationSpeed;
+				direction = -1;
```
`direction` is already set to `-1` at this point.

pchote commented:
Fixed and rebased.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/58286871ee8df5ff7b7156b2cd4f4d4f92235b5b...f29a67c6d647cb3ebba797af664349e57b53816b)
